app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -225,30 +236,25 @@ fun HomeScreen(
     BottomNavigationMenu(
         onTabSelected = { tab -> navigationActions.navigateTo(tab) },
         tabList = TOP_LEVEL_DESTINATIONS,
-        selectedItem = TOP_LEVEL_DESTINATIONS[2],
+        selectedItem = getTopLevelDestination(Route.HOME),
         modifier =
             Modifier.testTag("bottomNavMenu").constrainAs(bottomNav) {
               bottom.linkTo(parent.bottom)
               start.linkTo(parent.start)
               end.linkTo(parent.end)
             })
-
-    var viewToggleIcon =
-        if (viewToggleBrowseIndex == 0) Icons.Default.Place else Icons.AutoMirrored.Filled.List
-    FloatingActionButton(
+    ViewToggleFab(
+        modifier =
+            Modifier.padding(16.dp).testTag("viewToggleFab").constrainAs(viewToggle) {
+              bottom.linkTo(bottomNav.top)
+              absoluteRight.linkTo(parent.absoluteRight)
+            },
         onClick = {
-          viewToggleBrowseIndex = if (viewToggleBrowseIndex == 0) 1 else 0
+          uiState.viewList = !uiState.viewList
```

Comment:
Same here, to change the uiState, you will need access to `_uiState` in the viewModel. I see that there's no compilation error because you made the variable mutable (using `var`). It should not be possible to directly edit the UI state of the view model (it breaks the MVVM architecture). Instead, make everything a `val` and use helper functions to modify it. See examples in the bootcamp if needed. 

I see @christinehahaa also did this for her changes, so it would be good to sync up with her as well.